core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -336,13 +356,36 @@ private ClientTransport obtainActiveTransport() {
     @Override
     public <ReqT, RespT> ClientCall<ReqT, RespT> newCall(MethodDescriptor<ReqT, RespT> method,
         CallOptions callOptions) {
-      return new ClientCallImpl<ReqT, RespT>(
+      final ClientCallImpl<ReqT, RespT> call = new ClientCallImpl<ReqT, RespT>(
           method,
           new SerializingExecutor(executor),
           callOptions,
           transportProvider,
           scheduledExecutor)
               .setUserAgent(userAgent);
+      return new SimpleForwardingClientCall<ReqT, RespT>(call) {
+        @Override
+        public void start(ClientCall.Listener<RespT> responseListener, Metadata headers) {
+          activeCalls.add(call);
+          try {
+            super.start(new SimpleForwardingClientCallListener<RespT>(responseListener) {
+              @Override
+              public void onClose(Status status, Metadata trailers) {
+                try {
+                  super.onClose(status, trailers);
+                } finally {
+                  activeCalls.remove(call);
+                  checkTerminated();
```

Comment:
Hmm... this can cause multiple calls to `onChannelTerminated()` if two calls complete at the same time.

Hmm... This acquires a channel-wide lock per-call. We will probably want to remove that, but that can happen later.